src/librustc_parse/parser/path.rs

```diff
@@ -71,7 +71,23 @@ impl<'a> Parser<'a> {
             debug!("parse_qpath: (decrement) count={:?}", self.unmatched_angle_bracket_count);
         }
 
-        self.expect(&token::ModSep)?;
+        let lo_colon = self.token.span;
```

Comment:
Why is this necessary; `self.prev_span` should point at the colon (a single token) without needing `.to(...)`?

Reply:
Yes, you're right - I was being overly cautious about span handling